src/composite/MegaChart.js

```diff
@@ -52,41 +53,110 @@
+            this._chart.chartType(this.chartType());
+        }
+        
+        this.setLegend();
+        this._contentClasses = this.getContentClasses();
+        this.removeExtraLegends();
         
-            this._layout
-                .getContent("center")
-                    .columns(this.columns())
-                    .data(this.data());
-            
-            if(this._layout.getContent("center").chartType() !== this.chartType()){
-                this._layout.getContent("center").chartType(this.chartType());
+        this.setAxisTitles();
+        
+        if(this._legend !== null){
+            if(["top","bottom"].indexOf(this.legendPosition()) !== -1){
+                this._legend.orientation("horizontal");
+            } else {
+                this._legend.orientation("vertical");
             }
         }
-        if(this.showLegend()){
-            this._layout.setContent("right", new Legend().fixedSize(true).targetWidget(this._layout.getContent("center")));
-        } else {
-            this._layout.clearContent("right");
-        }
+        this._legend.dataType(this._chart.getChartDataType());
     };
+    
     MegaChart.prototype.exit = function (domNode, element) {
         Surface.prototype.exit.apply(this, arguments);
     };
+    
+    MegaChart.prototype.setLegend = function(){
+        if(this.showLegend() && (this._layout.getContent(this.legendPosition()) === null || this._layout.getContent(this.legendPosition()).classID() !== "other_Legend")){
+            this._legend = new Legend().fixedSize(true).targetWidget(this._layout.getContent("center"));
+            this._layout.setContent(this.legendPosition(), this._legend);
+        } else if (!this.showLegend()) {
+            this._layout.clearContent(this.legendPosition());
+        }
+    };
+    
+    MegaChart.prototype.setAxisTitles = function () {
+        if(this.valueAxisTitle() && this._contentClasses.left !== "common_Text"){
+            if(this.showLegend()){
+                if(this.legendPosition() !== "left"){
+                    this._layout.setContent("left", this._valueTitle.rotation(-90));
+                }
+            } else {
+                this._layout.setContent("left", this._valueTitle.rotation(-90));
+            }
+        }
+        if(this.domainAxisTitle() && this._contentClasses.bottom !== "common_Text"){
+            if(this.showLegend()){
+                if(this.legendPosition() !== "bottom"){
+                    this._layout.setContent("bottom", this._domainTitle);
+                }
+            } else {
+                this._layout.setContent("bottom", this._domainTitle);
+            }
+        }
+    };
+    
+    MegaChart.prototype.removeExtraLegends = function () {
+        for(var i in this._contentClasses){
+            if(this._contentClasses[i] === "other_Legend" && i !== this.legendPosition()){
+                this._layout.clearContent(i);
+            }
+        }
+    };
```

Comment:
Ok this addresses my previous comment.

I would prefer is you simply used a "prevLegendPosition" pattern:

```
if (this._prevLegendPosition !== this.legendPosition()) {
  //remove old and reposition it to new location (or create a new one)
  this._prevLegendPosition = this.legendPosition();
}
```
Also you could do that inside the update method.  

Thought:  You could remove the "showLegend PP completely and include a legendPosition of "none"?